File: problemset/230B.py
'''
    author:  AliTitan051
    Email: AliGhanbariCs@gmail.com
    GitHub: https://github.com/AliBinary
    created: 10.09.2023 14:11:22
'''

n = 10**6
A = [1]*n
B = set()
for x in range(2, n):
    if A[x]:
        B.add(x*x)
        for m in range(x, n, x):
            A[m] = 0

input()
for k in map(int, input().split()):
    print(["NO", "YES"][k in B])

File: problemset/230B.cpp
/*
    author:  Ali
    Email: AliGhanbariCs@gmail.com
    GitHub: https://github.com/AliBinary
    created: 06.02.2024 17:40:18
*/

#include <bits/stdc++.h>
using namespace std;

#define ll long long
#define vi vector<int>

bool isprime(int n)
{
    // corner case
    if (n <= 1)
        return false;

    // check from 2 to n/2
    for (int i = 2; i * i <= n; ++i)
        if (n % i == 0)
            return false;

    return true;
}

int main()
{
    ios::sync_with_stdio(0), cin.tie(0), cout.tie(0);

    int n;
    cin >> n;

    ll x;
    while (n--)
    {
        cin >> x;
        if ((ll)sqrt(x) != sqrt(x))
        {
            cout << "NO\n";
            continue;
        }
        x = sqrt(x);

        if (isprime(x))
            cout << "YES\n";
        else
            cout << "NO\n";
    }
}